app/services/CheckoutService.scala

```diff
@@ -276,7 +276,17 @@ class CheckoutService(identityService: IdentityService,
 
     def addPlan(contact: Contact) = {
       val newRatePlan = RatePlan(renewal.plan.id.get, None)
-      val renewCommand = Renew(subscription.id.get, subscription.termStartDate, subscription.termEndDate, NonEmptyList(newRatePlan), contractEffective, customerAcceptance)
+
+      val renewCommand = Renew(
+        subscriptionId = subscription.id.get,
+        currentTermStartDate = subscription.termStartDate,
+        currentTermEndDate = subscription.termEndDate,
+        newRatePlans = NonEmptyList(newRatePlan),
+        contractEffectiveDate = contractEffective,
+        customerAcceptanceDate = customerAcceptance,
+        promoCode = None,
+        autoRenew = renewal.plan.charges.billingPeriod.isInstanceOf[RecurringPeriod])
```

Comment:
not quite, I think this code is doing a type check, wheras pattern matching checks at compile time if it's a valid check and whether it's exhaustive (assuming it's a sealed trait)
both are valid types of polymorphism, but one is inheritence and the other is extending from outside